arrow/src/compute/kernels/comparison.rs

```diff
@@ -898,6 +899,224 @@ pub fn gt_eq_utf8_scalar<OffsetSize: StringOffsetSizeTrait>(
     compare_op_scalar!(left, right, |a, b| a >= b)
 }
 
+macro_rules! dyn_compare_scalar {
+    ($LEFT: expr, $RIGHT: expr, $OP: ident) => {{
+        let right: i128 = $RIGHT.try_into().map_err(|_| {
+            ArrowError::ComputeError(String::from("Can not convert scalar to i128"))
+        })?;
+        match ($LEFT.data_type(), $RIGHT.get_arrow_type()) {
+            // (DataType::Boolean, DataType::Boolean) => {
+            //     typed_cmp_scalar!($LEFT, $RIGHT, BooleanArray, $OP_BOOL)
+            // }
+            (DataType::Int8, DataType::Int8) => {
+                let right: i8 = right.try_into().map_err(|_| {
+                    ArrowError::ComputeError(String::from(
+                        "Can not convert scalar to i128",
+                    ))
+                })?;
+                let left =
+                    $LEFT.as_any().downcast_ref::<Int8Array>().ok_or_else(|| {
+                        ArrowError::CastError(String::from("Left array cannot be cast"))
+                    })?;
```

Comment:
I think you can use https://docs.rs/arrow/6.4.0/arrow/array/fn.as_primitive_array.html to simplify this

So something like

```suggestion
                    as_primitive_array::<Int8Array>($LEFT)?;
```

Reply:
I think these functions require an `Arc<dyn Array>` /`ArrayRef` which would mean we need the end user to do that before passing the array to the kernel or do you think its okay to add the `Arc` in the kernel?